src/compile/mark.ts

```diff
@@ -69,13 +69,13 @@ export function compileMark(model: Model): any[] {
         );
 
       return [{
-        name: (name ? name + '-' : '') + mark + '-facet',
+        name: (specName ? specName + '-' : '') + mark + '-facet',
```

Comment:
This one is very different from the other ones. Can you extract this into a helper function of the model?